packages/core/src/enums/ImageStatus.ts

```diff
@@ -0,0 +1,21 @@
+/** Status of a frame as it gets loaded */
+enum ImageStatus {
+  // Replicate is a duplicated image, from some larger distance
+  REPLICATE = 1,
+
+  // Skipping a value here and after the next replicate to allow for interpolation
+  // enum values.
+
+  // Adjacent replicate is a duplicated image of a nearby image
+  ADJACENT_REPLICATE = 3,
+  // Loading is used to prevent replication when the actual images start becoming available
+  LOADING = 5,
+  // Partial images
+  PARTIAL = 6,
+  // Lossy images, either complete or partial
+  LOSSY = 7,
+  // Done means the image is full resolution/complete
+  DONE = 8,
+}
```

Comment:
This enum is a bit unusual as it has two parts: one for image attributes such as replicate, lossy, partial, etc., and then there is LOADING. The loading is unusually placed here compared to others that are quality. I would even call this `ImageQualityStatus`.

Reply:
Or `ImageCompletionStatus` if you don't want to say quality, but in any case LOADING is a little bit unusual, you can say UNDEFINED? UNDECIDED? something that doesn't represent load, UNKNOWN?

Reply:
As discussed the loading should not be here as well